qa/sc_evm_forger.py

```diff
@@ -0,0 +1,426 @@
+        assert_equal(ft_amount_in_wei, initial_balance)
+
+        # try spending the stake by a sc node which does not own it
+        forg_spend_res_2 = sc_node_2.transaction_spendForgingStake(
+            json.dumps({"stakeId": str(stakeId_genesis)}))
+        assert_true('error' in forg_spend_res_2, "The command should fail")
+        assert_equal(forg_spend_res_2['error']['description'], "Forger Stake Owner not found")
+
+        # SC1 Delegate 300 Zen and 200 Zen to SC node 2 - expected stake is 500 Zen
+        sc2_blockSignPubKey = sc_node_2.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        sc2_vrfPubKey = sc_node_2.wallet_createVrfSecret()["result"]["proposition"]["publicKey"]
+
+        forgerStake1_amount = 300  # Zen
+        forgerStakes = {"forgerStakeInfo": {
+            "ownerAddress": evm_address_sc_node_1,  # SC node 1 is an owner
+            "blockSignPublicKey": sc2_blockSignPubKey,  # SC node 2 is a block signer
+            "vrfPubKey": sc2_vrfPubKey,
+            "value": convertZenToZennies(forgerStake1_amount)  # in Satoshi
+        }
+        }
+        makeForgerStakeJsonRes = sc_node_1.transaction_makeForgerStake(json.dumps(forgerStakes))
+        if "result" not in makeForgerStakeJsonRes:
+            fail("make forger stake failed: " + json.dumps(makeForgerStakeJsonRes))
+        else:
+            print("Forger stake created: " + json.dumps(makeForgerStakeJsonRes))
+        self.sc_sync_all()
+
+        # Generate SC block on SC node (keep epoch)
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=False)
+        self.sc_sync_all()
+        print_current_epoch_and_slot(sc_node_1)
+
+        stakeList = sc_node_1.transaction_allForgingStakes()["result"]['listOStakes']
+        assert_equal(len(stakeList), 2)
+
+        forgerStake2_amount = ft_amount_in_zen - forgerStake1_amount  # Zen
+        forgerStakes = {"forgerStakeInfo": {
+            "ownerAddress": evm_address_sc_node_1,  # SC node 1 is an owner
+            "blockSignPublicKey": sc2_blockSignPubKey,  # SC node 2 is a block signer
+            "vrfPubKey": sc2_vrfPubKey,
+            "value": convertZenToZennies(forgerStake2_amount)  # in Satoshi
+        }
+        }
+        makeForgerStakeJsonRes = sc_node_1.transaction_makeForgerStake(json.dumps(forgerStakes))
+        if "result" not in makeForgerStakeJsonRes:
+            fail("make forger stake failed: " + json.dumps(makeForgerStakeJsonRes))
+        else:
+            print("Forger stake created: " + json.dumps(makeForgerStakeJsonRes))
+        self.sc_sync_all()
+
+        # Generate SC block on SC node (keep epoch)
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=False)
+        self.sc_sync_all()
+        print_current_epoch_and_slot(sc_node_1)
+
+        # we now have 3 stakes
+        stakeList = sc_node_1.transaction_allForgingStakes()["result"]['listOStakes']
```

Comment:
I think about adding the api endpoint `allActiveForgingStakes`, that will return only those who can participate in forging. Can be useful even for this test.
Can be done as a separate task in the next sprints.

Reply:
Ok